### PR TITLE
New command 'attr_type'

### PR DESCRIPTION
**Pull request by t-wissmann**
The command only prints the type of a given attribute and will be of use
in the python bindings.

### CODE REVIEW

**Comment by tiosgz** [comment]
What about `attr_type` or `get_attr_type`?

**Comment by t-wissmann** [comment]
I tried to keep it consistent with `set_attr` and `get_attr`. Often `get` is redundant in command or function names. Let's see what @blarz thinks :)

**Comment by blarz** [comment]
> I tried to keep it consistent with `set_attr` and `get_attr`. Often `get` is redundant in command or function names. Let's see what @blarz thinks :)

I'll have a look tomorrow :)

File: doc/herbstluftwm.txt [hunk 1]
```diff
@@ -850,6 +850,9 @@ set_attr 'ATTRIBUTE' 'NEWVALUE'::
     Assign 'NEWVALUE' to the specified 'ATTRIBUTE' as described in the
     <<OBJECTS,*OBJECTS section*>>.
 
+type_of_attr 'ATTRIBUTE'::
+    Print (one line containing) the type of the specified 'ATTRIBUTE'.
```
Comment (blarz) [inline doc/herbstluftwm.txt:854]: Is the (one line containing) necessary? It kind of reads "bumpy", if you know what I mean

**Comment by blarz** [comment]
> I tried to keep it consistent with `set_attr` and `get_attr`. Often `get` is redundant in command or function names. Let's see what @blarz thinks :)

What about removing `of_` and reduce it to `type_attr`?
Would be consistent with the other `WORD_attr` (although type is not a verb ;))
Looks OK IMHO. What you think @t-wissmann and @tiosgz?

**Comment by tiosgz** [comment]
> What about removing `of_` and reduce it to `type_attr`?

Then it looks like an adjective to me (that is, an attribute storing a type). Or…
> (although type is not a verb ;))

…unfortunately, it is (`hc type_out_this_attr my_random_attr` into the terminal I’m running if from :P).

File: doc/herbstluftwm.txt (hunk 1 above)
Comment (t-wissmann) [reply doc/herbstluftwm.txt:854]: Hm, I think I can also remove it. It will be important that the output is precisely one line :)

**Comment by t-wissmann** [comment]
Hm, I think `type_attr` sounds like "typing the attribute" (i.e. assigning a type or typing it on a keyboard).

**Comment by blarz** [comment]
I see both your points :D

Not sure, `type_of_attr` seems.. too long or too verbose?

`attr_type` is snappy, I like it, too. But I also understand that it is not consistent with the other `*_attr` commands.

**Comment by tiosgz** [comment]
I could only think of ‘identify’ & ‘recognise’, which are both too vague or completely off.

Wrt. consistency… ‘get’ & ‘set’ are verbs, but so far we’ve got no verb to put there. When it comes to adjectives/‘of’ in other commands:
- `monitor_rect` not `rect_of_monitor`
- `object_tree` not `tree_of_objects`
- `tag_status` not `status_of_tags`

**Comment by t-wissmann** [comment]
I see your point. I've renamed it to `attr_type`. It's a pitty that this clashes with the tab-completion for `attr`, but at least typing `hc at`<kbd>Tab</kbd> will still complete to `hc attr` and the user only has to type the space manually.

**Comment by blarz** [comment]
> I see your point. I've renamed it to `attr_type`. It's a pitty that this clashes with the tab-completion for `attr`, but at least typing `hc at`Tab will still complete to `hc attr` and the user only has to type the space manually.

attr_type sounds good to me. What do you think @tiosgz ?

File: doc/herbstluftwm.txt (hunk 1 above)
Comment (blarz) [reply doc/herbstluftwm.txt]: Thanks! Reads better now IMHO. If it is important we can add it, maybe the only thing making it "bumpy" were the parantheses

**Comment by tiosgz** [comment]
> attr_type sounds good to me. What do you think @tiosgz ?

Sure it can be.

File: doc/herbstluftwm.txt (hunk 1 above)
Comment (tiosgz) [reply doc/herbstluftwm.txt]: ‘a’ instead of ‘one’ would also make it read better (IMO)

Comment (t-wissmann) [reply doc/herbstluftwm.txt]: I've removed the addition in the parentheses already.

Comment (blarz) [reply doc/herbstluftwm.txt]: Closing this now :)

**Review by blarz — APPROVED** [review]
In it goes :)

**Pull request merged**